src/magicbox/SuggestionsManager.ts

```diff
@@ -197,45 +169,104 @@ export class SuggestionsManager {
-        $$(dom).removeClass(this.options.selectedClass);
-        var found = $$(dom).find('.' + this.options.selectableClass);
-        $$(found).removeClass(this.options.selectedClass);
-      }
+
+    const suggestionsContainer = this.buildSuggestionsContainer();
+    $$(this.element).append(suggestionsContainer.el);
+
+    each(suggestions, (suggestion: Suggestion) => {
+      const dom = suggestion.dom ? this.modifyDomFromExistingSuggestion(suggestion.dom) : this.createDomFromSuggestion(suggestion);
+
+      dom.setAttribute('id', `magic-box-suggestion-${indexOf(suggestions, suggestion)}`);
+      dom.setAttribute('role', 'option');
+
       dom['suggestion'] = suggestion;
-      this.element.appendChild(dom);
+      suggestionsContainer.append(dom.el);
+    });
+
+    this.hasSuggestions = suggestions.length > 0;
+
+    $$(this.element).toggleClass('magic-box-hasSuggestion', this.hasSuggestions);
+    $$(this.magicBoxContainer).setAttribute('aria-expanded', this.hasSuggestions.toString());
+  }
+
+  private buildSuggestionsContainer() {
+    return $$('div', {
+      id: 'coveo-magicbox-suggestions',
+      role: 'listbox'
     });
-    if (suggestions.length > 0) {
-      $$(this.element).addClass('magic-box-hasSuggestion');
-      this.hasSuggestions = true;
+  }
+
+  private createDomFromSuggestion(suggestion: Suggestion) {
+    const dom = $$('div', {
+      className: `magic-box-suggestion ${this.options.selectableClass}`
+    });
+
+    dom.on('click', () => {
+      suggestion.onSelect();
+    });
+
+    dom.on('keyboardSelect', () => {
+      suggestion.onSelect();
+    });
+
+    if (suggestion.html) {
+      dom.el.innerHTML = suggestion.html;
+      return dom;
+    }
+
+    if (suggestion.text) {
+      dom.text(suggestion.text);
+      return dom;
+    }
+
+    if (suggestion.separator) {
+      dom.addClass('magic-box-suggestion-seperator');
+      const suggestionLabel = $$(
+        'div',
+        {
+          className: 'magic-box-suggestion-seperator-label'
+        },
+        suggestion.separator
+      );
+      dom.append(suggestionLabel.el);
+      return dom;
+    }
+
+    return dom;
+  }
+
+  private modifyDomFromExistingSuggestion(dom: HTMLElement) {
+    // this need to be done if the selection is in cache and the dom is set in the suggestion
+    $$(dom).removeClass(this.options.selectedClass);
+    const found = $$(dom).find('.' + this.options.selectableClass);
+    $$(found).removeClass(this.options.selectedClass);
+    return $$(dom);
+  }
+
+  private move(direction: 'up' | 'down') {
+    const currentlySelected = $$(this.element).find(`.${this.options.selectedClass}`);
+    const selectables = $$(this.element).findAll(`.${this.options.selectableClass}`);
+    const currentIndex = indexOf(selectables, currentlySelected);
+
+    let index = direction == 'up' ? currentIndex - 1 : currentIndex + 1;
+    if (index < -1) {
```

Comment:
Should this condition be `< 0` instead? Seems in the case when we reach the end of the list, we reset to `0` on line 256.

Reply:
No, we want that when we go out of bound by one, we reset to what is inside the combobox, not "wrap around" immediately.

You can see the same behaviour on google for example, if you go down one suggestion, then up 2 times, it's going to reset to the content of the combobox, not wrap around.